### PR TITLE
Upgrade amazon-lambda-http to use API Gateway 2.0 http event

### LINKED ISSUE #12115: Support version 2.0 of API Gateway Json
**Description**
Support version 2.0 of API Gateway Json illustrated here:

https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html

We currently only support 1.0

### PR DESCRIPTION
**Pull request by patriot1burke**
Implementation for: #12115

### CODE REVIEW

**Comment by patriot1burke** [comment]
FYI, documentation was updated as well to reflect much easier usage of 'sam' command and deployment.

**Comment by patriot1burke** [comment]
This is a big PR mainly because I kept the old v1 version around just in case people need it.  This is an important PR as the AWS docs on v1 weren't good and v2 seems to be a lot more flushed out and a little easier to configure and deploy.

**Comment by patriot1burke** [comment]
@gsmet @gastaldi @stuartwdouglas If you can't review, know somebody that can?

File: docs/src/main/asciidoc/amazon-lambda-http.adoc [hunk 1]
```diff
@@ -107,116 +111,58 @@ http://127.0.0.1:3000/hello
...
 [source,bash,subs=attributes+]
 ----
-aws cloudformation delete-stack --stack-name <YOUR_STACK_NAME>
+./mvnw clean install -Dnative
 ----
 
-== Execute your REST Lambda on AWS
-
-To get the root URL for your service, type the following command and see the following output:
+You can then test the executable locally with sam local
```
Comment (gastaldi) [inline docs/src/main/asciidoc/amazon-lambda-http.adoc:148]:
```suggestion
You can then test the executable locally with `sam local`
```

File: docs/src/main/asciidoc/amazon-lambda-http.adoc [hunk 2]
```diff
@@ -107,116 +111,58 @@ http://127.0.0.1:3000/hello
...
+In general, client code will automatically handle this encoding but in certain custom situations, you should be aware
+you may need to manually manage that encoding.
 
-The above error is stating that my S3 bucket should be in `us-east-2`, not `us-east-1`.
-To fix this error you'll need to create an S3 bucket in that region and redo steps 1 and 2 from above.
+== Deploying a native executable
 
-Another annoying this is that if there is an error in deployment, you also have to completely delete
-it before trying to deploy again:
+To deploy a native executable, you must build it with Graal.
```
Comment (gastaldi) [inline docs/src/main/asciidoc/amazon-lambda-http.adoc:141]:
```suggestion
To deploy a native executable, you must build it with GraalVM.
```

**Review by gastaldi — APPROVED** [review]
Added some comments, but looks good overall.

On another topic maybe it would be interesting to consider having the v1 API in a Quarkiverse extension instead?

**Comment by gastaldi** [comment]
Or rather, move the Amazon extensions to a `quarkus-amazon` repository in quarkiverse?

**Comment by stuartwdouglas** [comment]
Does this really need to be two different extensions? Is there a reason why you can't just select the legacy one with a build time config property, to avoid all the copying?

**Comment by patriot1burke** [comment]
> Does this really need to be two different extensions? Is there a reason why you can't just select the legacy one with a build time config property, to avoid all the copying?

There is no copying.  Its different code, same class names, different handlers, etc.  The json format is a bit different.  Smaller jar, less code.  Much clearner to hav a different extension we can deprecate and remove.  I also want to see if anybody notices!

**Pull request merged**